src/AbstractPhpdocTypesFixer.php

```diff
@@ -56,6 +56,7 @@ public function isCandidate(Tokens $tokens)
      */
     protected function applyFix(\SplFileInfo $file, Tokens $tokens)
     {
+        $tokensForAnalysis = clone $tokens;
```

Comment:
this is very expensive, if you iterate backwards;

```php
    /**
     * {@inheritdoc}
     */
    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
    {
        for ($i = \count($tokens) - 1; $i >= 0; --$i) {
            if (!$tokens[$i]->isGivenKind(T_DOC_COMMENT)) {
                continue;
            }

            $doc = new DocBlock($tokens[$i]->getContent());
            $annotations = $doc->getAnnotationsOfType($this->tags);

            if (empty($annotations)) {
                continue;
            }

            foreach ($annotations as $annotation) {
                $this->fixTypes($tokens, $annotation);
            }

            $tokens[$i] = new Token([T_DOC_COMMENT, $doc->getContent()]);
        }
    }
```

all tests pass. However, I'm not sure why you are cloning here so maybe this doesn't work?

Reply:
Well, I've started with touching `\PhpCsFixer\AbstractPhpdocTypesFixer` as minimum as possible, and without your fix the loop gets infinite because `\PhpCsFixer\Tokenizer\Resolver\TypeShortNameResolver` calls (somewhere in its dependencies) `\PhpCsFixer\Tokenizer\TokensAnalyzer` which resets the cursor with `$this->tokens->rewind();`

I'll apply your fix soon :+1: